src/dsp_tools/analyse_xml_data/construct_and_analyze_graph.py

```diff
@@ -188,73 +188,75 @@ def _remove_edges_to_stash(
         remaining_nodes: indices of the nodes in the graph
-
-    Returns:
-        the links that should be stashed
     """
-    source, target = edges_to_remove[0].source_rx_index, edges_to_remove[0].target_rx_index
-    links_to_stash = [x.link_object for x in edges_to_remove]
-    # if only one (source, target) is entered, it removes only one edge, not all
-    # therefore we need as many entries in the list as there are edges between the source and target to break the cycle
-    to_remove_list = [(x.source_rx_index, x.target_rx_index) for x in edges_to_remove]
-    phantom_links = []
-    for link_to_stash in links_to_stash:
+    # find the normal edges to remove
+    normal_edges_to_remove = [(x.source_rx_index, x.target_rx_index) for x in edges_to_remove]
+    # if only one (source, target) is removed, it removes only one edge, not all
+    # therefore we need as many entries in the list as there are edges between the source and the target
+
+    # find the phantom edges to remove
+    phantom_edges_to_remove = []
+    for link_to_stash in [x.link_object for x in edges_to_remove]:
         if isinstance(link_to_stash, XMLLink):
-            phantom_links.extend(_find_phantom_xml_edges(source, target, all_edges, link_to_stash, remaining_nodes))
-    to_remove_list.extend(phantom_links)
-    graph.remove_edges_from(to_remove_list)
-    return links_to_stash
+            new_phantoms = _find_phantom_xml_edges(
+                edges_to_remove[0].source_rx_index,
+                edges_to_remove[0].target_rx_index,
+                all_edges,
+                link_to_stash,
+                remaining_nodes,
+            )
+            phantom_edges_to_remove.extend(new_phantoms)
+
+    # remove the normal edges and the phantom edges
+    all_edges_to_remove = normal_edges_to_remove + phantom_edges_to_remove
+    graph.remove_edges_from(all_edges_to_remove)
 
 
 def _find_phantom_xml_edges(
-    source: int,
-    target: int,
-    edges: list[Edge],
-    xml_instance: XMLLink,
+    source_rx_index: int,
+    target_rx_index: int,
+    all_edges: list[Edge],
+    xml_link_to_stash: XMLLink,
     remaining_nodes: set[int],
 ) -> list[tuple[int, int]]:
     """
-    If an edge that will be removed represents an XML link
-    the link may contain further links to other resources.
-    If we stash the XMLLink then in the real data all the links are stashed.
-    This is not automatically the case in the rx graph.
-    We identify all the edges that need to be removed so that the rx graph represents the links that remain
-    in the real data.
+    If an edge that will be removed represents an XML link,
+    the text value may contain further links to other resources.
+    If we stash the XMLLink, then in the real data all links of that text value are stashed.
+    So, the rx graph must be purged from these "phantom" links that are not in the real data any more.
+    This function identifies the edges that must be removed from the rx graph.
```

Comment:
```suggestion
    This function identifies these "phantom" links, so they can be removed from the graph.
```
I don't like the phrasing "purge", also "rx" is not relevant here.